src/shims/fs.rs

```diff
@@ -330,68 +411,44 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         // If the `AT_SYMLINK_NOFOLLOW` flag is set, we query the file's metadata without following
         // symbolic links.
-        let metadata = if flags & this.eval_libc("AT_SYMLINK_NOFOLLOW")?.to_i32()? != 0 {
-            // FIXME: metadata for symlinks need testing.
-            std::fs::symlink_metadata(path)
-        } else {
-            std::fs::metadata(path)
-        };
-
-        let metadata = match metadata {
-            Ok(metadata) => metadata,
-            Err(e) => {
-                this.set_last_error_from_io_error(e)?;
-                return Ok(-1);
-            }
-        };
+        let is_symlink = flags & this.eval_libc("AT_SYMLINK_NOFOLLOW")?.to_i32()? != 0;
```

Comment:
"is_symlink" is a bad variable name. This doesn't determine whether the file is a symlink. This determines where resolution should *follow* the symlink or not.

Reply:
I changed it to `follow_symlink`.